src/components/StackNameGenerator.jsx

```diff
@@ -16,29 +16,39 @@ class StackNameGenerator extends React.Component {
     this.makeRequest = this.makeRequest.bind(this);
   }
 
-  async makeRequest(letter) {
+  async makeRequest(letter, currentList) {
     return axios({
       method: "get",
       url: this.URL + letter,
       responseType: "text"
     }).then(async response => {
-      let results = response.data.split("\n");
-      let filteredResults = results.filter(word => word != null && word !== "");
-      const randomIndex = Math.floor(Math.random() * filteredResults.length);
-      return filteredResults[randomIndex];
+      const results = this.dedupeLoop(1,response.data.split('\n').filter(Boolean),currentList);
+      return results[Math.floor(Math.random() * results.length)];
     });
   }
 
+  dedupeLoop(loopCount, possibleStrings, currentList) {
```

Comment:
attempts to find values based on how many times the list has been iterated; fir example, first iteration will only grab values that have been used < 1 times; second iteration, less than 2, and so on; basically ensuring that when duplicates are encountered, a value will not be repeated until all available items have been exhausted for that iteration... furthermore, if you have 16 A's in the name and 4 possible A values, you are guaranteed to see each value exactly 4 times